CMakeLists.txt

```diff
@@ -25,6 +25,13 @@ if(POLICY CMP0077)
   cmake_policy(SET CMP0077 NEW)
 endif()
 
+# Enable Swift_COMPILATION_MODE property.
+set(CMP0157_IS_NEW FALSE)
+if(POLICY CMP0157)
+  cmake_policy(SET CMP0157 NEW)
+  set(CMP0157_IS_NEW TRUE)
```

Comment:
Instead of setting this variable, we should probably use `cmake_policy(GET...)` at the locations where we are querying for it so that it can't get out of sync and there's only one source of truth.

Reply:
I didn't use `cmake_policy(GET...)` because it would require a temporary variable.
```cmake
cmake_policy(GET CMP0157 CMP0157_value)
if(NOT CMP0157_value STREQUAL NEW)
  ...
endif()
```
Is there a way to check the value in `if(...)` condition?